code/ai/aicode.cpp

```diff
@@ -7747,41 +7746,36 @@ void ai_choose_secondary_weapon(object *objp, ai_info *aip, object *en_objp)
 
 		if (is_big_ship)
 		{
-			wif_priority1 = WIF_HUGE;
-			wif_priority2 = (aip->ai_profile_flags & AIPF_SMART_SECONDARY_WEAPON_SELECTION) ? WIF_BOMBER_PLUS : WIF_HOMING;
-			wif2_priority1 = WIF2_CAPITAL_PLUS;
-			wif2_priority2 = 0;
+            wif_priority1 += Weapon::Info_Flags::Huge;
+            if (aip->ai_profile_flags & AIPF_SMART_SECONDARY_WEAPON_SELECTION) {
+                wif_priority2 += Weapon::Info_Flags::Bomber_plus;
+            }
+            else {
+                wif_priority2 = wif_priority2 + Weapon::Info_Flags::Homing_aspect + Weapon::Info_Flags::Homing_heat + Weapon::Info_Flags::Homing_javelin;
```

Comment:
Yet another instance of repeatedly adding flags (and another a few lines below).